src/Chapter1.hs

```diff
@@ -653,7 +674,12 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: (Integral a) => a -> a
+firstDigit = go . abs
```

Comment:
That's elegant! This way, you do `abs` only once, outside the recursion function. Amazing job! 👏🏼

Reply:
@JeremyBois  It depends on the case, I guess. I could see that it may be helpful when you use some lazy-strict tricks, but in general, I think it is okay not to do so until compared for the performance 🙂

Reply:
Ok I will follow your advice 👍